src/main/asciidoc/_chapters/ops_mgt.adoc

```diff
@@ -2475,14 +2471,14 @@ When nodes are removed from the slave cluster, or if nodes go down or come back
 
 ==== Keeping Track of Logs
 
-Each master cluster region server has its own znode in the replication znodes hierarchy.
-It contains one znode per peer cluster (if 5 slave clusters, 5 znodes are created), and each of these contain a queue of WALs to process.
+Before 3.0.0, for zookeeper based implementation, it is like a tree, we have a znode for a peer cluster, but under the znode we have lots of files.
+But after 3.0.0, for table based implementation, we have server name in row key, which means we will have lots of rows for a given peer.
 Each of these queues will track the WALs created by that region server, but they can differ in size.
 For example, if one slave cluster becomes unavailable for some time, the WALs should not be deleted, so they need to stay in the queue while the others are processed.
 See <<rs.failover.details,rs.failover.details>> for an example.
 
 When a source is instantiated, it contains the current WAL that the region server is writing to.
-During log rolling, the new file is added to the queue of each slave cluster's znode just before it is made available.
+During log rolling, the new file is added to the queue of each slave cluster's record just before it is made available.
```

Comment:
This is different for table based replication queue storage, and it is the key point here. For zookeeper, it is an external system so there is no problem to let log rolling depend on it, but if we want to store the state in a hbase table, we can not let log rolling depend on it as it will introduce dead lock...
We will only write to hbase:replication when want to record an offset after replicating something.